### ISSUE TITLE
Snapshot releases (Mac and Linux versions) on MacOS are not properly loading

### ISSUE THREAD

**Issue by trnstlntk** [OpenRefine/OpenRefine#4727]
When opening the Mac OS (.dmg) snapshot release, it doesn't load OpenRefine any further than the logo, as seen in this screenshot:
![image](https://user-images.githubusercontent.com/9881645/162605130-f818968a-eed4-44df-9499-273de95bfdf2.png)

When I try with the Linux download, that one doesn't even open a new browser tab at all.

### Expected Behavior
OpenRefine should fully load.

### Context
Terminal log when trying to start the MacOS .dmg (pretty minimal):
```
Sandras-MacBook-Air:~ fokky$ /Applications/OpenRefine\ snapshot\ 20220408.app/Contents/MacOS/JavaAppLauncher ; exit;
08:15:29.649 [            refine_server] Starting Server bound to '127.0.0.1:3333' (0ms)
08:15:29.717 [            refine_server] Initializing context: '/' from '/Applications/OpenRefine snapshot 20220408.app/Contents/Resources/webapp' (68ms)
08:15:31.931 [                   refine] Starting OpenRefine 3.6-SNAPSHOT [0ba3b92]... (2214ms)
08:15:31.937 [                   refine] initializing FileProjectManager with dir (6ms)
08:15:31.937 [                   refine] /Users/fokky/Library/Application Support/OpenRefine (0ms)
```

When I try to open the Linux version of the snapshot release, that one doesn't even start up a new browser tab. Terminal log for that one is:

```
miniMac:~ fokky$ /Users/fokky/Desktop/openrefine-3.6-SNAPSHOT\ 3/refine ; exit;
Using refine.ini for configuration
cat: refine.ini: No such file or directory
-------------------------------------------------------------------------------------------------
You have 812M of free memory.
Your current configuration is set to use 1024M of memory.
OpenRefine can run better when given more memory. Read our FAQ on how to allocate more memory here:
https://docs.openrefine.org/manual/installing#increasing-memory-allocation
-------------------------------------------------------------------------------------------------
ls: server/target/lib: No such file or directory
\nCould not find Maven locally, starting download for Maven ...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 8469k  100 8469k    0     0  10.7M      0 --:--:-- --:--:-- --:--:-- 10.7M
/Users/fokky/Desktop/openrefine-3.6-SNAPSHOT 3/refine: line 294: cd: main/webapp: No such file or directory
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.300 s
[INFO] Finished at: 2022-04-10T09:26:43+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] The goal you specified requires a project to execute but there is no POM in this directory (/Users/fokky/Desktop). Please verify you invoked Maven from the correct directory. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MissingProjectException
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.426 s
[INFO] Finished at: 2022-04-10T09:26:47+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] The goal you specified requires a project to execute but there is no POM in this directory (/Users/fokky/Desktop). Please verify you invoked Maven from the correct directory. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MissingProjectException


Error: Could not find or load main class com.google.refine.Refine
Caused by: java.lang.ClassNotFoundException: com.google.refine.Refine
logout

Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

[Process completed]
```

### Versions
 - Operating System: MacOS 10.13.6, MacOS 12.2.1
 - Browser Version: Chrome
 - JRE or JDK Version: n/a?
 - OpenRefine: 3.6 snapshot release

### Files used

I just tried with the April 8 and April 10 snapshot releases, but the problem has been around for longer (several weeks?).

I heard from a Linux user (via the Structured Data on Commons Telegram group) that they have the same issue of OpenRefine snapshot releases not loading for them.

**Comment by trnstlntk** [comment]
I have just downloaded today's (April 13, 2022) snapshot releases for MacOS (.dmg) and for Linux.

The MacOS one starts up correctly! 💃

But the Linux one not yet (tried on two macs - MacOS 12.3.1 and MacOS 10.13.6).

The terminal produces the following:

```
miniMac:~ fokky$ /Users/fokky/Desktop/openrefine-3.6-SNAPSHOT\ 20220413/refine ; exit;
Using refine.ini for configuration
cat: refine.ini: No such file or directory
-------------------------------------------------------------------------------------------------
You have 472M of free memory.
Your current configuration is set to use 1024M of memory.
OpenRefine can run better when given more memory. Read our FAQ on how to allocate more memory here:
https://docs.openrefine.org/manual/installing#increasing-memory-allocation
-------------------------------------------------------------------------------------------------
ls: server/target/lib: No such file or directory
/Users/fokky/Desktop/openrefine-3.6-SNAPSHOT 20220413/refine: line 294: cd: main/webapp: No such file or directory
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.199 s
[INFO] Finished at: 2022-04-13T16:26:45+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] The goal you specified requires a project to execute but there is no POM in this directory (/Users/fokky/Desktop). Please verify you invoked Maven from the correct directory. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MissingProjectException
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.174 s
[INFO] Finished at: 2022-04-13T16:26:47+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] The goal you specified requires a project to execute but there is no POM in this directory (/Users/fokky/Desktop). Please verify you invoked Maven from the correct directory. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MissingProjectException


Error: Could not find or load main class com.google.refine.Refine
Caused by: java.lang.ClassNotFoundException: com.google.refine.Refine
logout

Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.
Deleting expired sessions...6 completed.

[Process completed]
```

**Comment by elebitzero** [comment]
Hey @trnstlntk,  I tested the latest snapshot builds for Linux (openrefine-3.6-20220413.185658-237-linux.tar) and Windows (openrefine-3.6-20220413.185658-237-win-with-java) and they are working.

Command line output for Ubuntu 21.10:

```
$ ./refine run
Using refine.ini for configuration
-------------------------------------------------------------------------------------------------
You have 5936M of free memory.
Your current configuration is set to use 1400M of memory.
OpenRefine can run better when given more memory. Read our FAQ on how to allocate more memory here:
https://docs.openrefine.org/manual/installing#increasing-memory-allocation
-------------------------------------------------------------------------------------------------

15:36:33.965 [            refine_server] Starting Server bound to '[127.0.0.1:3333](http://127.0.0.1:3333/)' (0ms)
15:36:33.989 [            refine_server] refine.memory size: 1400M JVM Max heap: 1419116544 (24ms)
15:36:34.112 [            refine_server] Initializing context: '/' from '/home/granite/software/openrefine-3.6-SNAPSHOT/webapp' (123ms)
Gtk-Message: 15:36:35.011: Failed to load module "canberra-gtk-module"
Gtk-Message: 15:36:36.012: Failed to load module "canberra-gtk-module"
Gtk-Message: 15:36:36.017: Failed to load module "canberra-gtk-module"
15:36:36.888 [                   refine] Starting OpenRefine 3.6-SNAPSHOT [dba2b8b]... (2776ms)
15:36:36.888 [                   refine] initializing FileProjectManager with dir (0ms)
15:36:36.888 [                   refine] /home/granite/.local/share/openrefine (0ms)
15:36:43.360 [                   refine] POST /command/core/load-language (6472ms)
```

Can you please redownload the latest snapshot, test on Linux, and report back if you continue to have issues?

**Comment by thadguidry** [comment]
@trnstlntk Your first few lines in your output look puzzling to me for launching from the refine script.  And then what follows is that it seems it tried to BUILD ?  Almost as if the `20220413/refine` default was to perform a build and not instead to execute the `./refine` in the directory?

**Comment by antoine2711** [comment]
> @trnstlntk Your first few lines in your output look puzzling to me for launching from the refine script. And then what follows is that it seems it tried to BUILD ? Almost as if the `20220413/refine` default was to perform a build (or the previous `3/refine` ) and not instead to execute the `./refine` in the directory?

@thadguidry:
what I see is:
`/Users/fokky/Desktop/openrefine-3.6-SNAPSHOT\ 20220413/refine ; exit;`

This is ONE directory: `openrefine-3.6-SNAPSHOT\ 20220413` because of the `\`. It escape the space, so I don't see any issues there.

This intriges me more… `cat: refine.ini: No such file or directory`
Would a bad base dir have impact on other parts of the execution?

Regards, Antoine

**Comment by trnstlntk** [comment]
I don't have a Linux machine/installation. I use MacOS but sometimes run the Linux build there. Admittedly, that's a small use case and I assume not very many people do that (most will run the .dmg).

I have just downloaded the latest Linux snapshot release (April 13, 18:56) - same issue there.

This is the log produced by my very old MacBook Air (which can't run a more recent OS than 10.13.6):

Maybe other Mac users can test (@antoine2711 ??) and see if they have this issue too?
As I said, I think I'm bumping upon a rare use case here.

**Comment by antoine2711** [comment]
> Maybe other Mac users can test (@antoine2711 ??) and see if they have this issue too? As I said, I think I'm bumping upon a rare use case here.

@trnstlntk: yes, I can test it. I have actually installed since 2 weeks 2 virtual machines (VirtualBox), one Ubuntu 21.10 and one Windows10. Since I'm on a Mac, I obviously also have a Mac, so I could also test on all three OS.

That being said, I always run from the master that I compile locally, so this is new to me. I will test it. Almost did today! ;-)

Regards, Antoine

P.S. for those who want to try, I could help you setup your Ubuntu virtualbox machine. It's not that complicated, though a lot of downloading is required! ;-)

**Comment by trnstlntk** [comment]
FYI, I'm not even running a VM. The way I run the Linux version on Mac is: download OpenRefine's Linux tarball, unzip, right click the `refine` file, tell it to run in the Terminal, tada.

![image](https://user-images.githubusercontent.com/9881645/163433202-90531ea7-04df-47be-8f31-21c3d7d6adc6.png)

**Comment by antoine2711** [comment]
- Works on Ubuntu v21.10
  ![image](https://user-images.githubusercontent.com/12217525/163512515-522c59a1-e260-4561-9474-b157b1810140.png)

- Works on Mac v11.6.5
  ![image](https://user-images.githubusercontent.com/12217525/163513998-e3edbdae-b175-439f-817c-aa5b82f8adba.png)
  
- Works on Windows 10
   ![image](https://user-images.githubusercontent.com/12217525/163514706-f330b155-2491-40ed-b2b3-bc7c45761f18.png)

- Works on Windows 10 + JAVA
  ![image](https://user-images.githubusercontent.com/12217525/163514774-46155adf-3de2-4ea6-b58c-b4cb746c85e7.png)

Works fine on all 4 platforms on my side.

Regards, Antoine

**Comment by trnstlntk** [comment]
> Works fine on all 4 platforms on my side.

Fascinating !!!

Double-checked here on MacOS 12.3.1 on my late 2014 Mac Mini...

Linux tarball for 3.5.2 opened in Mac Terminal: loads absolutely fine
![image](https://user-images.githubusercontent.com/9881645/163708999-f27154f6-2481-46e8-b59c-39e6e592058c.png)
Terminal log for this (successful 3.5.2) one:
```
miniMac:~ fokky$ /Users/fokky/Desktop/openrefine-3.5.2/refine ; exit;
Using refine.ini for configuration
WARNING: OpenRefine is not tested and not recommended for use with Java versions greater than 12.
-------------------------------------------------------------------------------------------------
You have 163M of free memory.
Your current configuration is set to use 1024M of memory.
OpenRefine can run better when given more memory. Read our FAQ on how to allocate more memory here:
https://docs.openrefine.org/manual/installing#increasing-memory-allocation
-------------------------------------------------------------------------------------------------

11:37:44.501 [            refine_server] Starting Server bound to '127.0.0.1:3333' (0ms)
11:37:44.505 [            refine_server] refine.memory size: 1024M JVM Max heap: 1073741824 (4ms)
11:37:44.622 [            refine_server] Initializing context: '/' from '/Users/fokky/Desktop/openrefine-3.5.2/webapp' (117ms)
11:37:47.831 [                   refine] Starting OpenRefine 3.5.2 [e3efd4e]... (3209ms)
11:37:47.831 [                   refine] initializing FileProjectManager with dir (0ms)
11:37:47.832 [                   refine] /Users/fokky/Library/Application Support/OpenRefine (1ms)
11:38:01.807 [                   refine] POST /command/core/load-language (13975ms)
11:38:01.854 [                   refine] GET /command/core/get-preference (47ms)
11:38:01.895 [                   refine] POST /command/core/load-language (41ms)
11:38:01.915 [                   refine] POST /command/core/load-language (20ms)
11:38:01.928 [                   refine] POST /command/core/load-language (13ms)
11:38:02.589 [                   refine] POST /command/core/get-importing-configuration (661ms)
11:38:02.615 [                   refine] GET /command/core/get-all-project-tags (26ms)
11:38:02.671 [                   refine] GET /command/core/get-all-project-metadata (56ms)
11:38:03.347 [                   refine] GET /command/core/get-csrf-token (676ms)
11:38:03.394 [                   refine] GET /command/core/get-languages (47ms)
11:38:03.494 [                   refine] GET /command/core/get-version (100ms)
11:38:03.735 [                   refine] GET /command/database/saved-connection (241ms)
```

And then with the 3.6 snapshot release of April 16, 2022 - this one does not open a new tab with attempt to run OR in my browser:
```
miniMac:~ fokky$ /Users/fokky/Desktop/openrefine-3.6-SNAPSHOT\ 20220416/refine ; exit;
Using refine.ini for configuration
cat: refine.ini: No such file or directory
-------------------------------------------------------------------------------------------------
You have 355M of free memory.
Your current configuration is set to use 1024M of memory.
OpenRefine can run better when given more memory. Read our FAQ on how to allocate more memory here:
https://docs.openrefine.org/manual/installing#increasing-memory-allocation
-------------------------------------------------------------------------------------------------
ls: server/target/lib: No such file or directory
/Users/fokky/Desktop/openrefine-3.6-SNAPSHOT 20220416/refine: line 294: cd: main/webapp: No such file or directory
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.112 s
[INFO] Finished at: 2022-04-17T11:41:57+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] The goal you specified requires a project to execute but there is no POM in this directory (/Users/fokky/Desktop). Please verify you invoked Maven from the correct directory. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MissingProjectException
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.104 s
[INFO] Finished at: 2022-04-17T11:41:59+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] The goal you specified requires a project to execute but there is no POM in this directory (/Users/fokky/Desktop). Please verify you invoked Maven from the correct directory. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MissingProjectException


Error: Could not find or load main class com.google.refine.Refine
Caused by: java.lang.ClassNotFoundException: com.google.refine.Refine
logout

Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

[Process completed]
```

Not sure if this relevant at all, but re: the Java version I am running:
```
miniMac:~ fokky$ java --version
openjdk 17.0.2 2022-01-18
OpenJDK Runtime Environment (build 17.0.2+8-86)
OpenJDK 64-Bit Server VM (build 17.0.2+8-86, mixed mode, sharing)
```

Feel free to ping me if I can help with tweaking things and testing again. Maybe I just have something odd on my side??

**Comment by wetneb** [comment]
@trnstlntk could you check if this is still current? If so I would be tempted to accept that the Linux snapshot does not work on a MacOS machine - after all it is not designed for that! And now you have a way to run OpenRefine directly from `git` so this is perhaps not so needed anymore?

**Comment by antoine2711** [comment]
> miniMac:~ fokky$ /Users/fokky/Desktop/openrefine-3.6-SNAPSHOT\ 20220416/miniMac:~ fokky$  ; exit;
> Using refine.ini for configuration
> cat: refine.ini: No such file or directory

@trnstlntk: I think you are making a mistake by starting OR that way. You should `cd` in the folder first. OR's script are not build to be run from another folder. So in your case, do this:

```
miniMac:~ fokky$ cd /Users/fokky/Desktop/openrefine-3.6-SNAPSHOT\ 20220416/
miniMac:~ fokky$ ./refine
```

or Right-Click the enclosing folder, not the refine app file, and do this:
```
miniMac:~ openrefine-3.6-SNAPSHOT\ 20220416$ ./refine
```

Regards,
   Antoine

**Comment by antoine2711** [comment]
After investivating more, I see that this is less relevant on Mac or Linux, but would be fatal on Windows. I'm creating an issue for Windows: #4879

Regards, Antoine

**Comment by trnstlntk** [comment]
> @trnstlntk could you check if this is still current? If so I would be tempted to accept that the Linux snapshot does not work on a MacOS machine - after all it is not designed for that! And now you have a way to run OpenRefine directly from `git` so this is perhaps not so needed anymore?

I just tested it, and it works on both my Macs! Not sure where/how things got fixed, but hey 👯

**Comment by trnstlntk** [comment]
> @trnstlntk: I think you are making a mistake by starting OR that way. You should `cd` in the folder first. OR's script are not build to be run from another folder. So in your case, do this:

When I read this, I went 'oh duhhh' and felt very stupid 🤦‍♀️

BUT: I tested both ways (right-clicking the `refine` file and letting it open in Terminal, regardless of where I was in my file structure / and cd-ing in my Terminal to the right folder and then running `./refine` via command line) and both worked. So I have the impression that both methods are really fine.

In any case, seems like my exotic edge case seems invalid now, so I'll be happy to close this, unless someone still feels a strong passion about keeping this issue open.
